bokeh/core/properties.py

```diff
@@ -1600,7 +1600,7 @@ def set_from_json(self, obj, json, models=None):
 
 class DataSpec(Either):
     def __init__(self, typ, default, help=None):
-        super(DataSpec, self).__init__(String, Dict(String, Either(String, typ)), typ, default=default, help=help)
+        super(DataSpec, self).__init__(String, Dict(String, Either(String, Instance(HasProps), typ)), typ, default=default, help=help)
```

Comment:
@bryevdv You had suggested that this could be made more specific.  Specifically, `Instance(Transform)`.  I had wanted to do that as well but I ran into a error when I executed the code.  The error was: 

    NameError: name 'Transform' is not defined

I believe this is because of a circular import.